### PR TITLE
Bug 1136612 - [Stingray] expose isVisible and focus API for crash_report

### PR DESCRIPTION
**Pull request by sean2449**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=29bdcacaaec4)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=dad3baab7aa4)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e9fe07f54856)

File: tv_apps/smart-system/js/crash_reporter.js [hunk 1]
```diff
@@ -93,6 +96,7 @@ window.CrashReporter = (function() {
     };
 
     var button = null;
+    var systemBanner = new SystemBanner();
```
Comment (huchengtw-moz) [inline tv_apps/smart-system/js/crash_reporter.js]: Why do you want to move the creation of systemBanner to here? It looks the same and we don't need to do this change.

File: tv_apps/smart-system/js/system_banner.js [hunk 2]
```diff
@@ -1,7 +1,10 @@
 'use strict';
+/* global focusManager */
 
 (function(exports) {
 
+  var focusableBanners = [];
```
Comment (huchengtw-moz) [inline tv_apps/smart-system/js/system_banner.js]: Please write comments about this variable.

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=88c3a93b66b1)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=dc502ccb9022)

File: tv_apps/smart-system/js/crash_reporter.js [hunk 3]
```diff
@@ -69,8 +69,10 @@ window.CrashReporter = (function() {
                addEventListener('action', function onAction() {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/crash_reporter.js]: I'd prefer putting the "dot" right before `addEventListener` for better readability.

File: tv_apps/smart-system/js/crash_reporter.js [hunk 4]
```diff
@@ -166,13 +170,60 @@ window.CrashReporter = (function() {
...
+    return false;
+  }
+
+  /**
+   * Return z-index
+   */
+  function getOrder() {
+    var zIndex;
+    var dialog = document.getElementById('crash-dialog');
+    if (dialog && screen.classList.contains('crash-dialog')) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/crash_reporter.js]: We may reuse `isVisible()` method here.

File: tv_apps/smart-system/js/crash_reporter.js [hunk 5]
```diff
@@ -166,13 +170,60 @@ window.CrashReporter = (function() {
...
+  }
+
+  /**
+   * Focus send-report button if dialog is visible. If crash-reports-header is
+   * also visible, focus the actionButton in crash-reports-header
+   */
+  function focus() {
+    var dialog = document.getElementById('crash-dialog');
+    var dialogButton = document.getElementById('send-report');
+    if (dialogButton && screen.classList.contains('crash-dialog')) {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/crash_reporter.js]: same here.

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=aafa353b98ab)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f6b82486a7af)

File: tv_apps/smart-system/js/system_banner.js [hunk 6]
```diff
@@ -90,16 +97,40 @@
...
+      var zIndex = window.getComputedStyle(focusableBanners[0]).zIndex;
+      return zIndex === 'auto' ? 0 : zIndex;
+    }
+    return null;
+  };
+
+  SystemBanner.focus = function() {
+    var banner = focusableBanners[focusableBanners.length - 1];
+    if (banner) {
+      setTimeout(function() {
```
Comment (luke-chang) [inline tv_apps/smart-system/js/system_banner.js]: Please remove `setTimeout` due to bug 1139314.

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=179d6967df45)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8ad857dc2ff0)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4351c8a9297b)

**Comment by try-server-hook** [comment]
[sean2449 Sung Chiu (sean2449)](https://github.com/sean2449) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=6096a3a64ddf)

**Pull request merged**